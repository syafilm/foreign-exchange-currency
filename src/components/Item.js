import React, { Component } from 'react'

class Item extends Component {
  handleDelete = () => {
    const {
      item,
      handleDelete
    } = this.props

    const confirm = window.confirm("Are you sure want to delete this record ?")
    if (confirm) {
      const id = item.id
      handleDelete(id)
    } else {
      console.log('You pressed Cancel!')
    }
  }

  calculateCurrency = (item, currency) => {
    return Math.round((item.price * currency) * 100) / 100
  }

  render(){
    const {
      item,
      currency,
    } = this.props

    return(
      <div className="card my-3">
        <div className="card-header border-0 position-relative">
          <div className="row">
            <div className="col-3">
              <h5>
                {item.typeCurrency}
              </h5>
            </div>
            <div className="col-9 text-right">
              <h5>
                {this.calculateCurrency(item, currency)}
              </h5>
            </div>
          </div>
          <div className="row">
            <div className="col-12">
              <i>1 USD  = {item.typeCurrency} {this.calculateCurrency(item, currency)}</i>
            </div>
          </div>
          <button className="btn btn-delete" onClick={this.handleDelete}>
            <span className="d-block"> - </span>
          </button>
        </div>
      </div>
    )
  }
}

export default Item