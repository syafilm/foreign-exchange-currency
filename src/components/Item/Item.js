import React, { Component } from 'react'
import PropTypes from 'prop-types'

class Item extends Component {
  handleDelete = () => {
    const {
      item,
      handleDelete,
    } = this.props

    const confirm = window.confirm('Are you sure want to delete this record ?')
    if (confirm) {
      const { id } = item
      handleDelete(id)
    } else {
      console.log('You pressed Cancel!')
    }
  }

  calculateCurrency = (item, currency) => Math.round((item.price * currency) * 100) / 100

  render() {
    const {
      item,
      currency,
    } = this.props

    return (
      <div className="card my-3">
        <div className="card-header border-0 position-relative">
          <div className="row">
            <div className="col-3">
              <h5>
                {item.typeCurrency}
              </h5>
            </div>
            <div className="col-9 text-right">
              <h5>
                {this.calculateCurrency(item, currency)}
              </h5>
            </div>
          </div>
          <div className="row">
            <div className="col-12">
              <i>
                1 USD  =
                {item.typeCurrency}
                {' '}
                {this.calculateCurrency(item, currency)}
              </i>
            </div>
          </div>
          <button
            type="button"
            className="btn btn-delete"
            onClick={this.handleDelete}
          >
            <span className="d-block"> - </span>
          </button>
        </div>
      </div>
    )
  }
}

Item.propTypes = {
  item: PropTypes.object,
  currency: PropTypes.string,
  handleDelete: PropTypes.func,
}

Item.defaultProps = {
  item: {},
  currency: '100',
  handleDelete: null,
}

export default Item
