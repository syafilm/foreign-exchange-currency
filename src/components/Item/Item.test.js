
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Item from './ITem'

const item = {
  date: '2019-04-23',
  id: 'c0c66ac3-45d2-41e9-9e4d-769dcc9f4648',
  price: 0.7687861272,
  typeCurrency: 'GBP',
}

describe('Component Item', () => {
  it('Can render Item component smoothly', () => {
    const mockEvent = jest.fn()
    const currency = '100'
    const component = shallow(
      <Item
        key={item.id}
        item={item}
        handleDelete={mockEvent}
        currency={currency}
      />,
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
    expect(component.find('div')).toBeDefined()
  })
})
