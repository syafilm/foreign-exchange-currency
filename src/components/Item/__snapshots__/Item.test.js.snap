// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Item Can render Item component smoothly 1`] = `
<div
  className="card my-3"
>
  <div
    className="card-header border-0 position-relative"
  >
    <div
      className="row"
    >
      <div
        className="col-3"
      >
        <h5>
          GBP
        </h5>
      </div>
      <div
        className="col-9 text-right"
      >
        <h5>
          76.88
        </h5>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <i>
          1 USD  =
          GBP
           
          76.88
        </i>
      </div>
    </div>
    <button
      className="btn btn-delete"
      onClick={[Function]}
      type="button"
    >
      <span
        className="d-block"
      >
         - 
      </span>
    </button>
  </div>
</div>
`;
